File: todo-list/todo-connector.mjs
#!/usr/bin/env node

/**
 * Connector script for the Todo List skill
 * This script processes commands received from Clawdbot's /todo command
 */

import { execSync } from 'child_process';
import { join } from 'path';

// Get the command arguments passed from Clawdbot
const [, , ...args] = process.argv;

if (args.length === 0) {
  console.log(`
Todo List Manager

Usage:
  /todo add [--high|--low] [--due YYYY-MM-DD] [--category NAME] <item text>    Add a new todo
  /todo list [all|pending|completed] [--category NAME]                        List todos with optional filters
  /todo complete <id>                                                         Mark todo as complete
  /todo remove <id>                                                           Remove a todo
  /todo clear-completed                                                       Remove all completed todos
  /todo stats                                                                 Show statistics
  /todo categories                                                            List all categories
  /todo add-category <name>                                                   Add a new category
  /todo remove-category <name>                                                Remove a category
  /todo set-category <id> <category>                                          Assign todo to category
  `);
  process.exit(0);
}

try {
  // Execute the main todo script with the provided arguments
  const todoScriptPath = join(process.cwd(), 'skills/todo-list/todo.mjs');
  const command = `node "${todoScriptPath}" ${args.join(' ')}`;
  
  const result = execSync(command, { encoding: 'utf8' });
  console.log(result.trim());
} catch (error) {
  console.error(`Error executing todo command: ${error.message}`);
  process.exit(1);
}

File: todo-list/todo_connector.py
#!/usr/bin/env python3

import subprocess
import sys
from pathlib import Path


def main():
    args = sys.argv[1:]

    if not args:
        print('''Todo List Manager

Usage:
  /todo add [--priority high|medium|low] [--due YYYY-MM-DD] [--category NAME] [--assignee NAME] <item text>
  /todo list [--filter all|pending|completed] [--assignee NAME]
  /todo update <id> [--text "text"] [--priority high|medium|low] [--due YYYY-MM-DD] [--category NAME] [--assignee NAME] [--completed|--pending]
  /todo complete <id>
  /todo remove <id>
  /todo clear-completed
  /todo stats
  /todo categories
  /todo add-category <name>
  /todo remove-category <name>
  /todo set-category <id> <category>  (deprecated)
''')
        sys.exit(0)

    try:
        script_dir = Path.cwd() / 'skills' / 'todo-list'
        command = ['python', str(script_dir / 'todo.py')] + args

        result = subprocess.run(
            command,
            capture_output=True,
            text=True,
            encoding='utf-8'
        )
        print(result.stdout.strip())
        if result.stderr:
            print(result.stderr, file=sys.stderr)
    except Exception as e:
        print(f'Error executing todo command: {e}', file=sys.stderr)
        sys.exit(1)


if __name__ == '__main__':
    main()
